cron/builder/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	codeship "github.com/codeship/codeship-go"
+)
+
+// EnvKeyCSOrganization is the environment variable for
+// the Codeship organization
+const EnvKeyCSOrganization = "CS_ORGANIZATION"
+
+// EnvKeyCSPassword is the environment variable for
+// the Codeship password
+const EnvKeyCSPassword = "CS_PASSWORD"
+
+// EnvKeyCSUsername is the environment variable for
+// the Codeship username
+const EnvKeyCSUsername = "CS_USERNAME"
+
+// EnvKeyCSProjectUUID is the environment variable for
+// the uuid of the target Codeship project
+const EnvKeyCSProjectUUID = "CS_PROJECT_UUID"
+
+// EnvKeyCSBuildReference is the environment variable for
+// the desired reference for the build of the Codeship project
+// e.g. "20.04"  (which will have "heads/" tacked onto the beginning, if it isn't there)
+const EnvKeyCSBuildReference = "CS_BUILD_REFERENCE"
+
+func getRequiredString(envKey string, configEntry *string) error {
+	if *configEntry != "" {
+		return nil
+	}
+
+	value := os.Getenv(envKey)
+	if value == "" {
+		return fmt.Errorf("required value missing for environment variable %s", envKey)
+	}
+	*configEntry = value
+
+	return nil
+}
+
+func (c *BuilderConfig) init() error {
+	if err := getRequiredString(EnvKeyCSOrganization, &c.CSOrganization); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(EnvKeyCSPassword, &c.CSPassword); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(EnvKeyCSUsername, &c.CSUsername); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(EnvKeyCSProjectUUID, &c.CSProjectUUID); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(EnvKeyCSBuildReference, &c.CSBuildReference); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type BuilderConfig struct {
+	CSOrganization string `json:"CSOrganization"`
+	CSPassword     string `json:"CSPassword"`
+	CSUsername     string `json:"CSUsername"`
+
+	CSBuildReference string `json:"CSBuildReference"`
+	CSProjectUUID    string `json:"CSProjectUUID"`
+}
+
+func getProject(ctx context.Context, config BuilderConfig, org *codeship.Organization) (codeship.Project, error) {
```

Comment:
Would it be better to use the `GetProject` call? https://pkg.go.dev/github.com/codeship/codeship-go?utm_source=godoc#Organization.GetProject

Reply:
What is this function used by? I don't see any calls to it, nor do I see what part of this process needs information about the project yet.

Reply:
I'm not calling that function anywhere and just left it in there as an example.  Maybe I'll just delete it all together.